src/cargo/core/manifest.rs

```diff
@@ -41,7 +41,9 @@ impl EitherManifest {
 /// This is deserialized using the [`TomlManifest`] type.
 #[derive(Clone, Debug)]
 pub struct Manifest {
+    original: Rc<TomlManifest>,
     summary: Summary,
+
```

Comment:
Nit: Adding a comment for the two sections explaining what they are and why they are split up, could be helpful